docs/learn/prompts.md

```diff
@@ -1,219 +1,425 @@
-    """Now this is a normal docstring."""
-    
     genre: str
+
+
+prompt = BookRecommendationPrompt(genre="fantasy")
+print(prompt)
+# > Recommend a book.
+#   It should be a fantasy book.
 ```
 
-To call the model with a `BasePrompt`, create an instance of your class and invoke the `run()` method with the appropriate provider’s configuration decorator. The `run()` function will create a decorated function covered in more detail in the next section[link to calls].
+### Format Specifiers
 
-For the configuration decorator, you are required to set the `model` argument. You can also set  `stream` for streams[link] `response_model` for extractions[link], and `tools` for function calling/tools[link]. All other functionality is supported in the provider’s native API, and the original parameters can be passed directly as keyword arguments into the decorator.
+Since Mirascope prompt templates are just formatted strings, standard specifiers will work as expected:
 
 ```python
-from mirascope.core import openai, anthropic
+from mirascope.core import BasePrompt, prompt_template
 
-prompt = BookRecommendationPrompt(genre="fantasy")
 
-# Sync Call
-response = prompt.run(
-	openai.call(model="gpt-4o", temperature=1.0) # OpenAI takes a temperature flag
+@prompt_template("Recommend a book cheaper than ${price:.2f}")
+class BookRecommendationPrompt(BasePrompt):
+    price: float
+
+
+prompt = Prompt(price=12.3456)
+print(prompt)
+# > Recommend a book cheaper than $12.34
+```
+
+We also provide additional specifiers we've found useful in our own prompt engineering:
+
+- `list`: formats an input list as a newline-separated string
+- `lists`: formats an input list of lists as newline-separated strings separated by double newlines
+
+```python
+from mirascope.core import BasePrompt, prompt_template
+
+
+@prompt_template(
+    """
+    Recommend a book from one of the following genres:
+    {genres:list}
+
+    Examples:
+    {examples:lists}
+    """
 )
+class BookRecommendationPrompt(BasePrompt):
+    genres: list[str]
+    examples: list[tuple[str, str]]
+
 
-# Async Stream
-response = prompt.run_async(
-	anthropic.call(model="claude-3-5-sonnet-20240620", stream=True)
+prompt = BookRecommendationPrompt(
+    genres=["fantasy", "scifi", "mystery"],
+    examples=[
+        ("Title: The Name of the Wind", "Author: Patrick Rothfuss"),
+        ("Title: Mistborn: The Final Empire", "Author: Brandon Sanderson"),
+    ]
 )
+print(prompt)
+# > Recommend a book from one of the following genres:
+#   fantasy
+#   scifi
+#   mystery
+#
+#   Examples:
+#   Title: The Name of the Wind
+#   Author: Patrick Rothfuss
+#
+#   Title: Mistborn: The Final Empire
+#   Author: Brandon Sanderson
 ```
 
-`BasePrompt` supports every way to call a model that you would via a function decorator. For a full list of the functionalities, check out the `BasePrompt` section of the API reference[link].
+If there are any other such specifiers you would find useful, let us know!
```

Comment:
perhaps link to github?

Reply:
I was thinking about that, but there will already be a github link in the header of the docs and I didn't want to be linking and mentioning our issues in every single doc. I would almost rather just remove the ask than link everywhere. Thoughts?